cellbase-lib/src/main/java/org/opencb/cellbase/lib/impl/core/DataReleaseSingleton.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2015-2020 OpenCB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opencb.cellbase.lib.impl.core;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.changestream.ChangeStreamDocument;
+import com.mongodb.client.model.changestream.FullDocument;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.bson.Document;
+import org.opencb.cellbase.core.config.CellBaseConfiguration;
+import org.opencb.cellbase.core.config.SpeciesConfiguration;
+import org.opencb.cellbase.core.exception.CellBaseException;
+import org.opencb.cellbase.core.models.DataRelease;
+import org.opencb.cellbase.lib.db.MongoDBManager;
+import org.opencb.cellbase.lib.managers.CellBaseManagerFactory;
+import org.opencb.commons.datastore.mongodb.MongoDBCollection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public final class DataReleaseSingleton {
+
+    // {key = dbname, value = species}
+    private Map<String, String> speciesMap = new HashMap<>();
+    // {key = dbname, value = assembly}
+    private Map<String, String> assemblyMap = new HashMap<>();
+    // {key = dbname, value = lock}
+    private Map<String, ReentrantReadWriteLock> rwLockMap = new HashMap<>();
+
+    // {key = dbname, value = { key = release, value = { key = data, value = collection } } }
+    private Map<String, Map<Integer, Map<String, MongoDBCollection>>> cachedData = new HashMap<>();
```

Comment:
`Map<Map<Map<Collection>>>` ... map-inception! 

Jokes aside, having 4 maps where the "key" is "dbName" in all of them is weird and confusing. Wouldn't be better to have an internal private class `DatabaseInfo` with something like : 
```java
private static class DatabaseInfo {
 private final String dbName;
 private final String species;
 private final String assembly;
 private final ReentrantReadWriteLock rwLock;
 private final Map<Integer, Map<String, Collection>> cacheData;
}
```

Then, you just need a single `Map<String, DatabaseInfo>`